editor/src/components/canvas/canvas-strategies/flex-align-parent-strategy.ts

```diff
@@ -0,0 +1,260 @@
+import React from 'react'
+import { MetadataUtils } from '../../../core/model/element-metadata-utils'
+import { foldEither } from '../../../core/shared/either'
+import * as EP from '../../../core/shared/element-path'
+import {
+  ElementInstanceMetadata,
+  emptyComments,
+  jsxAttributeValue,
+  JSXElement,
+} from '../../../core/shared/element-template'
+import { setJSXValuesAtPaths, ValueAtPath } from '../../../core/shared/jsx-attributes'
+import {
+  canvasPoint,
+  CanvasPoint,
+  CanvasRectangle,
+  magnitude,
+  rectContainsPoint,
+} from '../../../core/shared/math-utils'
+import { ElementPath } from '../../../core/shared/project-file-types'
+import * as PP from '../../../core/shared/property-path'
+import {
+  EditorState,
+  forUnderlyingTargetFromEditorState,
+  modifyUnderlyingForOpenFile,
+  TransientFilesState,
+} from '../../editor/store/editor-state'
+import {
+  CanvasStrategy,
+  CanvasStrategyUpdateFnResult,
+  FlexAlignControlRectProps,
+  SelectModeCanvasSessionProps,
+  SelectModeCanvasSessionState,
+} from './canvas-strategy-types'
+
+export const flexAlignParentStrategy: CanvasStrategy = {
+  name: "Change Parent's Flex Align and Justify",
+  fitnessFn: (editor, currentSession) => {
+    if (editor.selectedViews.length === 1) {
+      const selectedView = editor.selectedViews[0]
+
+      const isFlexLayouted = MetadataUtils.isParentYogaLayoutedContainerForElementAndElementParticipatesInLayout(
+        MetadataUtils.findElementByElementPath(editor.jsxMetadata, selectedView),
+      )
+      const hasNoSiblings = MetadataUtils.getSiblings(editor.jsxMetadata, selectedView).length === 1
+
+      if (isFlexLayouted && hasNoSiblings) {
+        return 10 // fit!
+      }
+    }
+    return 0 // not fit
+  },
+  updateFn: (
+    lifecycle: 'transient' | 'final',
+    editorState: EditorState,
+    sessionProps: SelectModeCanvasSessionProps,
+    sessionState: SelectModeCanvasSessionState,
+  ): CanvasStrategyUpdateFnResult => {
+    // only apply after a certain treshold IF we hadn't already passed that treshold once
```

Comment:
This check should arguably be part of the fitness function

Reply:
thinking out loud from a user POV: if I press mouse down, do I want to immediately see the name of the active strategy, even if it has an internal rule to wait for a dragging treshold? or should we only enter the strategy once we are beyond that treshold? 
@seanparsons had an idea that the "wait for crossing treshold" could be either 1. it's own strategy, or 2. a "super-strategy" that wraps other strategies

Reply:
I'd argue that it would be best to wait until the threshold has been passed before communicating the active strategy, otherwise it gives the false (and potentially alarming) impression that the model is already being updated.
I quite like the idea of the "wait for crossing treshold" being its own strategy

Reply:
yeah I think I err on this side too, but I have an argument for immediately selecting a strategy on mouse down: once we have a strategy picker, we probably want to show the strategy picker ASAP, and not only after a threshold.

Reply:
I'm gonna leave the code as-is for now, will revisit in further PRs